Stop using currency/checkout zone/default country preferences

damianlegawiec opened the pull request:
All of it is set on Store-level, and shouldn't be set on Application-level anymore.

These preferences were always used as the last-resort fallback until now. We need to finally drop this deadweight and move on.

damianlegawiec committed:
Deprecate `Spree::Config[:currency']`

Currency is set in Store/Order, there shouldn't be a system override

damianlegawiec committed:
Deprecate `Spree::Config[:checkout_zone]`

damianlegawiec committed:
Deprecate `Spree::Config[:default_country_id]`

MatthewKennedy reviewed (approved):
This looks good, but the mysql ruby 3 tests seems unhappy for some reason.

damianlegawiec committed:
Deprecate `Country#ensure_not_default`

damianlegawiec committed:
Skip flaky test on MySQL

Generally, creating/deleting tables on the fly in tests is not a good idea :)

Replaced `database_cleaner` with `database_cleaner-active_record`
